src/main/java/gregtech/client/event/FluidVisualHandler.java

```diff
@@ -62,12 +61,11 @@ public static void onBlockOverlayRender(@Nonnull RenderBlockOverlayEvent event)
         final BlockPos blockpos = new BlockPos(player.posX, player.posY + player.getEyeHeight(), player.posZ);
         final Block block = player.world.getBlockState(blockpos).getBlock();
 
-        if (block instanceof MaterialFluidBlock) {
-            final MaterialFluidBlock fluidBlock = (MaterialFluidBlock) block;
-            final Material material = fluidBlock.getGTMaterial();
-            float r = ((material.getMaterialRGB() >> 16) & 0xFF) / 255.0F;
-            float g = ((material.getMaterialRGB() >> 8) & 0xFF) / 255.0F;
-            float b = (material.getMaterialRGB() & 0xFF) / 255.0F;
+        if (block instanceof GTFluidBlock fluidBlock) {
```

Comment:
Since GT Materials can set a custom color, do we want to check if the registered GTFluidBlock is from a GT Material first (again with the nullable gtmaterial option in FluidBlock), and then call the material's rgb color?